clients/githubrepo/branches.go

```diff
@@ -574,28 +606,32 @@ func mergeCheckRules(base, translated *clients.StatusChecksRule) {
 	}
 }
 
-func mergePullRequestReviews(base, translated *clients.PullRequestReviewRule) {
-	if readIntPtr(translated.RequiredApprovingReviewCount) > readIntPtr(base.RequiredApprovingReviewCount) {
-		base.RequiredApprovingReviewCount = translated.RequiredApprovingReviewCount
+func mergePullRequestReviews(base **clients.PullRequestReviewRule, translated *clients.PullRequestReviewRule) {
+	switch {
+	case translated == nil:
+		// "translated" have nothing to be merged in base
+		return
+	case *base == nil:
+		// result would be "translated" itself
+		*base = translated
+		return
 	}
-	if base.DismissStaleReviews == nil || readBoolPtr(translated.DismissStaleReviews) {
-		base.DismissStaleReviews = translated.DismissStaleReviews
+
+	if (*base).RequiredApprovingReviewCount == nil ||
+		valueOrDefault((*base).RequiredApprovingReviewCount, 0) < valueOrDefault(translated.RequiredApprovingReviewCount, 0) {
+		(*base).RequiredApprovingReviewCount = translated.RequiredApprovingReviewCount
 	}
-	if base.RequireCodeOwnerReviews == nil || readBoolPtr(translated.RequireCodeOwnerReviews) {
-		base.RequireCodeOwnerReviews = translated.RequireCodeOwnerReviews
+	if (*base).DismissStaleReviews == nil || valueOrDefault(translated.DismissStaleReviews, false) {
+		(*base).DismissStaleReviews = translated.DismissStaleReviews
 	}
-}
-
-func readBoolPtr(b *bool) bool {
-	if b == nil {
-		return false
+	if (*base).RequireCodeOwnerReviews == nil || valueOrDefault(translated.RequireCodeOwnerReviews, false) {
+		(*base).RequireCodeOwnerReviews = translated.RequireCodeOwnerReviews
 	}
-	return *b
 }
 
-func readIntPtr(i *int32) int32 {
-	if i == nil {
-		return 0
+func valueOrDefault[T any](ptr *T, defaultValue T) T {
+	if ptr == nil {
+		return defaultValue
 	}
-	return *i
+	return *ptr
 }
```

Comment:
+1 only saw this recently and it's a nice way of getting the zero value
```go
var zero T
return zero
```

>  I just want to make sure we won't want to use the valueOrDefault function with a default value other then the zero-value. I think this might be applicable to some case.

I had tried. to come up with a scenario where this was applicable, but Diogo had pointed out it wasn't working.

Reply:
same comment about moving the comment up for doc purposes:
```go
// returns the pointer's value if it exists, the type's zero-value otherwise
func readPtr[T any](ptr *T) T {
	if ptr == nil {
		var zero T
		return zero
	}
	return *ptr
}
```